### PR TITLE
Move policy functions to _timescaledb_functions schema

### PR DESCRIPTION
**Pull request by svenklemm**
To increase schema security we do not want to mix our own internal objects with user objects. Since chunks are created in the _timescaledb_internal schema our internal functions should live in a different dedicated schema. This patch make the necessary adjustments for the following functions:

- policy_compression_check(jsonb)
- policy_compression_execute(integer,integer,anyelement,integer,boolean,boolean)
- policy_compression(integer,jsonb)
- policy_job_error_retention_check(jsonb)
- policy_job_error_retention(integer,jsonb)
- policy_recompression(integer,jsonb)
- policy_refresh_continuous_aggregate_check(jsonb)
- policy_refresh_continuous_aggregate(integer,jsonb)
- policy_reorder_check(jsonb)
- policy_reorder(integer,jsonb)
- policy_retention_check(jsonb)
- policy_retention(integer,jsonb)

Disable-check: force-changelog-file

Disable-check: commit-count

### CODE REVIEW

File: tsl/test/expected/cagg_bgw.out [hunk 1]
```diff
@@ -697,6 +697,6 @@ SELECT count(*) FROM _timescaledb_config.bgw_job
    AND proc_name = 'policy_refresh_continuous_aggregate';
```
Comment (jnidzwetzki) [inline tsl/test/expected/cagg_bgw.out:697]: Line 696:

```
  WHERE proc_schema = '_timescaledb_functions'
```

File: tsl/test/expected/cagg_bgw_dist_ht.out [hunk 2]
```diff
@@ -734,7 +734,7 @@ SELECT count(*) FROM _timescaledb_config.bgw_job
    AND proc_name = 'policy_refresh_continuous_aggregate';
```
Comment (jnidzwetzki) [inline tsl/test/expected/cagg_bgw_dist_ht.out:734]: Line 733:

```
WHERE proc_schema = '_timescaledb_functions'
```

File: tsl/test/expected/compress_bgw_reorder_drop_chunks.out [hunk 3]
```diff
@@ -77,19 +77,19 @@ SELECT add_retention_policy('test_retention_table', INTERVAL '4 months') as rete
 SELECT count(*) FROM _timescaledb_config.bgw_job WHERE proc_schema = '_timescaledb_internal' AND proc_name = 'policy_retention';
```
Comment (jnidzwetzki) [inline tsl/test/expected/compress_bgw_reorder_drop_chunks.out:77]:
```suggestion
SELECT count(*) FROM _timescaledb_config.bgw_job WHERE proc_schema = '_timescaledb_functions' AND proc_name = 'policy_retention';
```

**Review by jnidzwetzki — APPROVED** [review]
Added a few comments about missing test changes, rest looks good to me.

**Review by akuzm — APPROVED** [review]

**Pull request merged**